src/types/refs.rs

```diff
@@ -83,17 +83,36 @@ impl<T: ScriptClass> Ref<T> {
         U: ScriptClass,
     {
         let inst = unsafe { (self.0 .0.instance as *const ISerializable).as_ref() }?;
-        inst.class()
-            .base_iter_with_self()
-            .any(|class| class.name() == CName::new(U::NAME))
-            .then(|| unsafe { mem::transmute(self) })
+        inst.is_a::<U>().then(|| unsafe { mem::transmute(self) })
     }
 
     /// Returns whether the reference is null.
     #[inline]
     pub fn is_null(&self) -> bool {
         self.0 .0.instance.is_null()
     }
+
+    #[inline]
+    pub fn is_exactly_a<U>(&self) -> bool
+    where
+        U: ScriptClass,
+    {
+        !self.is_null()
+            && unsafe { (self.0 .0.instance as *const ISerializable).as_ref() }
```

Comment:
```suggestion
            && unsafe { (self.0 .0.instance as *const ISerializable).as_ref() }
```
as_ref already does a null check